### PR TITLE
add CMD to api/mqtt to show it in ioBroker, etc.

### PR DESCRIPTION
**Pull request by MichaelDvP**
This shows the CMD-Type in api/device/values with empty/NAN value, so ioBroker can check. Sets right number/enum type in get_value_info Tested from tp1de.

Some other smaller changes:
- fix output for `api/device/circuit/values` (it was broken since dhw circuit addition)
- change order of return_not_found calls. Looked confusion when message was "cannot find schedule in value" Also use the devicename from F_() to make changes easier
- all user set names are <20 characters, Change analogsensor default name to `type_gpio` (e.g. `dig_out_16`, `adc_17`).
- scheduler/values returns empty json if no name is set or no schedule defined.

### CODE REVIEW

File: interface/src/project/validators.ts [hunk 1]
```diff
@@ -412,13 +424,30 @@ export const isGPIOUniqueValidator = (sensors: AnalogSensor[]) => ({
   }
 });
 
+export const uniqueAnalogNameValidator = (sensors: AnalogSensor[]) => ({
+  validator(rule: InternalRuleItem, n: string, callback: (error?: string) => void) {
+    if (n !== '' && sensors.find((as) => as.n === n)) {
```
Comment (proddy) [inline interface/src/project/validators.ts:429]: should we check for not being case-sensitive? compare lower() to lower() ?

Comment (MichaelDvP) [reply interface/src/project/validators.ts]: Yes, i add a commit.

**Comment by proddy** [comment]
very nice!

**Comment by mattreim** [comment]
Hello, 
with the new version I get these errors:

2024-07-15 14:18:43.059 WARNING 26: [command] Command failed: invalid command 'rftemp'
2024-07-15 14:18:43.059 ERROR 27: [emsesp] API call failed. invalid command (Error)
2024-07-15 14:18:56.886 WARNING 28: [command] Command failed: invalid command 'dewtemperature'
2024-07-15 14:18:56.886 ERROR 29: [emsesp] API call failed. invalid command (Error)
2024-07-15 14:19:00.879 WARNING 30: [command] Command failed: invalid command 'airhumidity'
2024-07-15 14:19:00.879 ERROR 31: [emsesp] API call failed. invalid command (Error)

**Comment by mattreim** [comment]
With their last version ([v23](https://github.com/emsesp/EMS-ESP32/pull/1843#issuecomment-2214872621)) everything still worked.

EMS-remote

**Comment by mattreim** [comment]
My `api/thermostat/commands` from the v23

{
  "[hc\u003Cn\u003E.]boost": "Boost",
  "[hc\u003Cn\u003E.]boosttime": "Boost Dauer",
  "[hc\u003Cn\u003E.]comforttemp": "Komforttemperatur",
  "[hc\u003Cn\u003E.]control": "Fernsteuerung",
  "[hc\u003Cn\u003E.]controlmode": "Kontrollmodus",
  "[hc\u003Cn\u003E.]cooling": "Kühlen",
  "[hc\u003Cn\u003E.]designtemp": "Auslegungstemperatur",
  "[hc\u003Cn\u003E.]dewoffset": "Taupunkt Differenz",
  "[hc\u003Cn\u003E.]dhwprio": "WW-Vorrang",
  "[hc\u003Cn\u003E.]ecotemp": "eco Temperatur",
  "[hc\u003Cn\u003E.]fastheatup": "schnelles Aufheizen",
  "[hc\u003Cn\u003E.]heatingtype": "Heizungstyp",
  "[hc\u003Cn\u003E.]heatoffdelay": "Ausschaltverzögerung Heizen",
  "[hc\u003Cn\u003E.]heatondelay": "Einschaltverzögerung Heizen",
  "[hc\u003Cn\u003E.]hpminflowtemp": "WP minimale Vorlauftemperatur",
  "[hc\u003Cn\u003E.]hpmode": "WP Modus",
  "[hc\u003Cn\u003E.]hpoperatingmode": "Wärmepumpe Betriebsmodus",
  "[hc\u003Cn\u003E.]instantstart": "Sofortstart",
  "[hc\u003Cn\u003E.]intoffset": "Korrektur interner Temperatur",
  "[hc\u003Cn\u003E.]manualtemp": "manuelle Temperatur",
  "[hc\u003Cn\u003E.]maxflowtemp": "max Vorlauftemperatur",
  "[hc\u003Cn\u003E.]minflowtemp": "min Vorlauftemperatur",
  "[hc\u003Cn\u003E.]mode": "Modus",
  "[hc\u003Cn\u003E.]nofrostmode": "Frostschutz Modus",
  "[hc\u003Cn\u003E.]nofrosttemp": "Frostschutztemperatur",
  "[hc\u003Cn\u003E.]noreducetemp": "Durchheizen unter",
  "[hc\u003Cn\u003E.]offsettemp": "Temperaturanhebung",
  "[hc\u003Cn\u003E.]program": "Programm",
  "[hc\u003Cn\u003E.]reducemode": "Absenkmodus",
  "[hc\u003Cn\u003E.]reducetemp": "Absenkmodus unter",
  "[hc\u003Cn\u003E.]remotehum": "Raumfeuchte Remote",
  "[hc\u003Cn\u003E.]remotetemp": "Raumtemperatur Remote",
  "[hc\u003Cn\u003E.]roominflfactor": "Raumeinflussfaktor",
  "[hc\u003Cn\u003E.]roominfluence": "Raumeinfluss",
  "[hc\u003Cn\u003E.]roomtempdiff": "Raumtemperatur Differenz",
  "[hc\u003Cn\u003E.]seltemp": "Sollwert Raumtemperatur",
  "[hc\u003Cn\u003E.]summersetmode": "Einstellung Sommerbetrieb",
  "[hc\u003Cn\u003E.]summertemp": "Sommertemperatur",
  "[hc\u003Cn\u003E.]switchonoptimization": "Einschaltoptimierung",
  "[hc\u003Cn\u003E.]tempautotemp": "temporäre Solltemperatur",
  "building": "Gebäudetyp",
  "commands": "Liste aller Kommandos",
  "damping": "Dämpfung der Außentemperatur",
  "datetime": "Datum/Zeit",
  "delayboiler": "Verzögerungs-Option",
  "dhw[n].charge": "Laden",
  "dhw[n].chargeduration": "Ladedauer",
  "dhw[n].circmode": "Zirkulationspumpen-Modus",
  "dhw[n].dailyheating": "täglich Heizen",
  "dhw[n].dailyheattime": "tägliche Heizzeit",
  "dhw[n].disinfectday": "Desinfektionstag",
  "dhw[n].disinfecting": "Desinfizieren",
  "dhw[n].disinfecttime": "Desinfektionszeit",
  "dhw[n].mode": "Modus",
  "dhw[n].settemp": "Solltemperatur",
  "dhw[n].settemplow": "untere Solltemperatur",
  "electricfactor": "Energiefaktor elektrisch",
  "energycostratio": "Energie/Kosten-Verhältnis",
  "entities": "Liste aller Entitäten",
  "fossilefactor": "Energiefaktor Fossil",
  "hybridstrategy": "Hybrid Strategie",
  "info": "Liste aller Werte",
  "intoffset": "Korrektur interner Temperatur",
  "minexttemp": "min. Aussentemperatur",
  "pvenabledhw": "aktiviere Anhebung WW",
  "pvlowercool": "Kühlabsenkung mit PV",
  "pvraiseheat": "Anhebung Heizen mit PV",
  "switchovertemp": "Außentemperatur für Umschaltung",
  "tempdiffboiler": "Temperaturdifferenz-Option"
}

**Comment by MichaelDvP** [comment]
Ah, indeed the value info for the second thermostat is broken.

**Comment by proddy** [comment]
> Ah, indeed the value info for the second thermostat is broken.

do you want me to take a look? I probably broke it.

**Comment by MichaelDvP** [comment]
fixed it, see PR

**Comment by mattreim** [comment]
Thanks

**Pull request merged**